api/net/ip4/udp.hpp

```diff
@@ -141,27 +150,52 @@ namespace net {
     public:
       Port_in_use_exception(UDP::port_t p)
         : port_(p) {}
-      virtual const char* what() const noexcept {
-        return "UDP port already in use";
-      }
 
-      UDP::port_t port(){
-        return port_;
-      }
+      virtual const char* what() const noexcept
+      { return "UDP port already in use"; }
+
+      UDP::port_t port()
+      { return port_; }
 
     private:
       UDP::port_t port_;
     };
 
   private:
+    static constexpr uint16_t exp_t_ {60 * 5};
 
-    downstream  network_layer_out_;
-    Stack&      stack_;
+    std::chrono::minutes        flush_interval_{5};
+    downstream                  network_layer_out_;
+    Stack&                      stack_;
     std::map<port_t, UDPSocket> ports_;
-    port_t      current_port_;
+    port_t                      current_port_;
 
     // the async send queue
     std::deque<WriteBuffer> sendq;
+
+    /** Error entries are just error callbacks and timestamps */
+    class Error_entry {
+    public:
+      Error_entry(UDP::error_handler cb) noexcept
```

Comment:
By moving the implementation of the constructor to the cpp you can avoid including `<rtc>` in the header (reducing the scope).